mesonbuild/interpreter/type_checking.py

```diff
@@ -678,7 +707,16 @@ def _convert_darwin_versions(val: T.List[T.Union[str, int]]) -> T.Optional[T.Tup
         ContainerTypeInfo(list, (str, File, CustomTarget, CustomTargetIndex, GeneratedList, ExtractedObjects, BuildTarget)),
         listify=True,
         default=[],
-    )
+    ),
+    *[a.evolve(deprecated='1.3.0', deprecated_message='This argument has never done anything in jar(), and should be removed')
+      for a in _LANGUAGE_KWS],
+]
+
+_SHARED_STATIC_ARGS: T.List[KwargInfo[T.List[str]]] = [
+    *[l.evolve(name=f'{l.name.split("_", 1)[0]}_static_args', since='1.3.0')
+      for l in _LANGUAGE_KWS],
+    *[l.evolve(name=f'{l.name.split("_", 1)[0]}_shared_args', since='1.3.0')
```

Comment:
It feels like this would be a lot more readable as:
```python
[l.evolve(name=l.name.replace('_', '_static_'), since='1.3.0') for l in _LANGUAGE_KWS]
```